File: static/js/lib/urls.js
// @flow
import { include } from "named-urls";
import qs from "query-string";

export const getNextParam = (search: string) => qs.parse(search).next || "/";

export const bulkReceiptCsvUrl = (hash: string) =>
  `/api/b2b/orders/${hash}/codes/`;

export const routes = {
  root: "/",
  catalog: "/catalog/",
  webinars: "/webinars/",
  enterprise: "/enterprise/",
  blog: "/blog/",
  dashboard: "/dashboard/",
  accountSettings: "/account-settings/",
  logout: "/logout/",

  // authentication related routes
  login: include("/signin/", {
    begin: "",
    password: "password/",
    forgot: include("forgot-password/", {
      begin: "",
      confirm: "confirm/:uid/:token/",
    }),
  }),

  register: include("/create-account/", {
    begin: "",
    confirm: "confirm/",
    confirmSent: "confirm-sent/",
    details: "details/",
    error: "error/",
    extra: "extra/",
    denied: "denied/",
  }),

  profile: include("/profile/", {
    view: "",
    update: "edit/",
  }),

  checkout: "/checkout/",

  ecommerceAdmin: include("/ecommerce/admin/", {
    index: "",
    coupons: "coupons/",
    deactivate: "deactivate-coupons/",
    processSheets: "process-coupon-assignment-sheets/",
  }),

  ecommerceBulk: include("/ecommerce/bulk/", {
    bulkPurchase: "",
    receipt: "receipt/",
  }),
  receipt: "receipt/:orderId/",

  account: include("/account/", {
    confirmEmail: "confirm-email",
  }),
};

File: b2b_ecommerce/urls.py
"""URLs for business to business ecommerce"""
from django.urls import path

from b2b_ecommerce.views import (
    B2BCheckoutView,
    B2BCouponView,
    B2BEnrollmentCodesView,
    B2BOrderStatusView,
)
from mitxpro.views import index


urlpatterns = [
    path("api/b2b/checkout/", B2BCheckoutView.as_view(), name="b2b-checkout"),
    path(
        "api/b2b/orders/<uuid:hash>/codes/",
        B2BEnrollmentCodesView.as_view(),
        name="b2b-enrollment-codes",
    ),
    path(
        "api/b2b/orders/<uuid:hash>/status/",
        B2BOrderStatusView.as_view(),
        name="b2b-order-status",
    ),
    path("api/b2b/coupon_status/", B2BCouponView.as_view(), name="b2b-coupon-view"),
    path("ecommerce/bulk/", index, name="bulk-enrollment-code"),
    path("ecommerce/bulk/receipt/", index, name="bulk-enrollment-code-receipt"),
]
